### PR TITLE
SPEC-2513 Fixing w4018

### PR DESCRIPTION
**Pull request by amznestebanpapp**
https://docs.microsoft.com/en-us/cpp/error-messages/compiler-warnings/compiler-warning-level-3-c4018?view=msvc-160

```
Using the token operator to compare signed and unsigned numbers required the compiler to convert the signed value to unsigned.
```

### CODE REVIEW

File: Code/Editor/ViewportTitleDlg.cpp [hunk 1]
```diff
@@ -451,19 +451,19 @@ void CViewportTitleDlg::AddFOVMenus(QMenu* menu, std::function<void(float)> call
     {
         for (size_t i = 0; i < customPresets.size(); ++i)
```
Comment (nemerle) [inline Code/Editor/ViewportTitleDlg.cpp:452]: maybe a replacement with ?
```cpp
        for (const QString & val : customPresets)
        {
            
        }        
```

File: Code/Editor/ViewportTitleDlg.cpp [hunk 2]
```diff
@@ -537,21 +537,21 @@ void CViewportTitleDlg::AddAspectRatioMenus(QMenu* menu, std::function<void(int,
 
     for (size_t i = 0; i < customPresets.size(); ++i)
     {
-        if (customPresets[i].isEmpty())
+        if (customPresets[static_cast<int>(i)].isEmpty())
```
Comment (nemerle) [inline Code/Editor/ViewportTitleDlg.cpp:540]: And same here, since `i` is not used in those loops

File: Code/Editor/ViewportTitleDlg.cpp [hunk 3]
```diff
@@ -668,21 +668,21 @@ void CViewportTitleDlg::AddResolutionMenus(QMenu* menu, std::function<void(int,
...
```
Comment (nemerle) [inline Code/Editor/ViewportTitleDlg.cpp:671]: ditto.

File: Code/Editor/ViewportTitleDlg.cpp (hunk 1 above)
Comment (amznestebanpapp) [reply Code/Editor/ViewportTitleDlg.cpp:452]: sure, replacing

File: Code/Editor/ViewportTitleDlg.cpp (hunk 2 above)
Comment (amznestebanpapp) [reply Code/Editor/ViewportTitleDlg.cpp:540]: sure, replacing

File: Code/Editor/ViewportTitleDlg.cpp (hunk 3 above)
Comment (amznestebanpapp) [reply Code/Editor/ViewportTitleDlg.cpp:671]: sure

**Review by lumberyard-employee-dm — APPROVED** [review]

**Review by AMZN-ScottR — APPROVED** [review]

**Pull request merged**